pytorch_lightning/strategies/hpu_parallel.py

```diff
@@ -0,0 +1,145 @@
+from pytorch_lightning.overrides.torch_distributed import broadcast_object_list
+from pytorch_lightning.plugins.io.checkpoint_plugin import CheckpointIO
+from pytorch_lightning.plugins.io.hpu_plugin import HPUCheckpointIO
+from pytorch_lightning.plugins.precision import PrecisionPlugin
+from pytorch_lightning.strategies.ddp import DDPStrategy
+from pytorch_lightning.utilities.distributed import group as _group
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _HPU_AVAILABLE, _TORCH_LESSER_EQUAL_1_10_2
+from pytorch_lightning.utilities.rank_zero import rank_zero_warn
+
+if _HPU_AVAILABLE:
+    import habana_frameworks.torch.core.hccl  # noqa: F401
+    from habana_frameworks.torch.utils.library_loader import load_habana_module
+
+log = logging.getLogger(__name__)
+
+
+class HPUParallelStrategy(DDPStrategy):
+    """Strategy for distributed training on multiple HPU devices."""
+
+    # The main process in each node spawns N-1 child processes via :func:`subprocess.Popen`, where N is the number of
+    # devices (e.g. GPU) per node. It is very similar to how :mod:`torch.distributed.launch` launches processes.
+
+    # Multi-device per process is not supported with habana
+
+    strategy_name = "hpu_parallel"
+
+    def __init__(
+        self,
+        accelerator: Optional["pl.accelerators.accelerator.Accelerator"] = None,
+        parallel_devices: Optional[List[torch.device]] = None,
+        checkpoint_io: Optional[CheckpointIO] = None,
+        precision_plugin: Optional[PrecisionPlugin] = None,
+        process_group_backend: Optional[str] = "hccl",
+    ) -> None:
+
+        if not _HPU_AVAILABLE:
+            raise MisconfigurationException("`HPUParallelStrategy` requires HPU devices to run")
+
+        super().__init__(
+            accelerator=accelerator,
+            parallel_devices=parallel_devices,
+            checkpoint_io=checkpoint_io or HPUCheckpointIO(),
+            precision_plugin=precision_plugin,
+            process_group_backend=process_group_backend,
+        )
+
+    def setup_environment(self) -> None:
+        # This function is used to load Habana libraries required for PyTorch
+        # to register HPU as one of the available devices.
+        load_habana_module()
+
+        os.environ["ID"] = str(self.local_rank)
+        os.environ["PL_TORCH_DISTRIBUTED_BACKEND"] = "hccl"
+
+        super().setup_environment()
+
+    def determine_ddp_device_ids(self) -> None:
+        return None
+
+    def pre_configure_ddp(self):  # type: ignore
+        # if unset, default `find_unused_parameters` `True`
+        # Many models require setting this parameter to True, as there are corner cases
+        # when not all parameter backward hooks are fired by the autograd engine even if require_grad is set to True.
+        # This flag does come with a performance hit, so it is suggested to disable in cases where it is possible.
```

Comment:
this should be moved to docstrings, not as random comment within the code :)

Reply:
This is inherited from DDP.